pkg/cloudprovider/cloud.go

```diff
@@ -140,6 +141,16 @@ type Instances interface {
 	// CurrentNodeName returns the name of the node we are currently running on
 	// On most clouds (e.g. GCE) this is the hostname, so we provide the hostname
 	CurrentNodeName(hostname string) (types.NodeName, error)
+	// InstanceExists returns true if the instance for the given node name still is running.
+	// If false is returned with no error, the instance will be immediately deleted.
+	// Note: The `cloudprovider.InstanceNotFound` error should not be returned here to notify
+	// the CCM that the instance is no longer running.
+	InstanceExists(nodeName types.NodeName) (bool, error)
+	// InstanceExistsByProviderID returns true if the instance for the given provider id still is running.
+	// If false is returned with no error, the instance will be immediately deleted.
+	// Note: The `cloudprovider.InstanceNotFound` error should not be returned here to notify
```

Comment:
@luxas do you think I should just omit that? I added it originally since people are use to using that error to denote the instance is gone, so I wanted to call out to not use that. However, calling it out in hindsight seems more confusing. Thoughts on just removing that note?